File: kotlin/dp/main_11726.kt
fun main() {
    val num = readLine()!!.toInt()
    var a = 0;
    var b = 1;
    for (i in 0.until(num)){
        val tem = a;
        a = b;
        b = (tem + b) % 10007;
    }
    println(b)
}

File: cpp/step/dp/main_11726.cpp
#include <iostream>

using namespace std;

int main() {
	int num;
	cin >> num;

	int a = 0;
	int b = 1;
	for (int i = 0; i < num; ++i) {
		int tem = a;
		a = b;
		b = (tem + b) % 10007;
	}
	cout << b << endl;
}
